### DISCUSSION TITLE
Adding a custom middleware to help with custom authentication

### DISCUSSION THREAD

**Discussion by semoac** [apache/airflow#23423] [Q&A]
Hi!

I'm trying to enable REMOTE_USER as authentication method and I need to add a new middleware to process a header.

I did find information about using `ADDITIONAL_MIDDLEWARE` to add the new middleware on superset, but looks like that no longer works.

Can someone help me by find information on how to define a middleware by editing  `webserver_config.py`? I'm kinda lost between flask appbuilder, airflow and superset.

Thanks.

**Comment by stijndehaes** [comment]
I did the following to loa the xray middle ware for flask:

```
from flask import current_app
from aws_xray_sdk.core import xray_recorder
from aws_xray_sdk.ext.flask.middleware import XRayMiddleware

xray_recorder.configure(service='My application')
XRayMiddleware(current_app, xray_recorder)
```
I did load it in an Airflow plugin instead of the webserver_config.py.

**Comment by slmg** [comment]
See https://github.com/apache/airflow/issues/32747
